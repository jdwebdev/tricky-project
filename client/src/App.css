/* APP */
.App {
    text-align: center;
}

.startBtnContainer {
    margin-top: 100px;
}

.startBtn {
    background-color: #282c34;
    width: 250px;
    height: 200px;
    border: none;
    border-radius: 10px;
    box-shadow: 10px 5px 5px #bcbcbc;
    color: white;
    font-size: 30px;
    cursor: pointer;
    transition: font-size 0.2s, color 0.2s, background-color 0.2s, width 0.2s,
        height 0.2s;
}

.startBtn:hover {
    background-color: rgb(106, 106, 106);
    color: black;
    font-size: 50px;
    width: 270px;
    height: 220px;
}
