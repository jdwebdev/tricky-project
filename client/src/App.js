import React from 'react';
import './App.css';
import Game from './components/Game';
import Introduction from './components/Introduction';
import { useState } from 'react'

const App = () => {

  const [gameState, setGameState] = useState('intro');

  const changeGameState = (newState) => {
    setGameState(newState);
  }

  return (
    <div className="App">
      <header>
          <h1>Tricky Project</h1>
      </header>
      {gameState === "intro" && 
        <Introduction 
          onClick={() => changeGameState("game")}
        />
      }
      {gameState === "game" &&
        <Game 
          finishGame={() => changeGameState("end")}
        />
      }
      {gameState === "end" && 
        <div>
          <p>GAME OVER</p>
          <p>ICI Affichage du score etc.</p>
        </div>
      }
    </div>
  );
}

export default App;
