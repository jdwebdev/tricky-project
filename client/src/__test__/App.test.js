/* eslint-disable no-undef */
import React from 'react'
import App from '../App'
import { shallow } from "enzyme"
import { shallowToJson } from "enzyme-to-json"


describe("app component", () => {
  it("renders App component without crashing", () => {
    shallow(<App />)
  })

  it("renders correctly", () => {
    const tree = shallow(<App />)
    expect(shallowToJson(tree)).toMatchSnapshot()
  })

  let appWrapper;
  beforeEach(() => {
    appWrapper = shallow(<App />)
  })

  it("h1 displays game title", () => {
    const title = appWrapper.find('h1').text()
    expect(title).toEqual("Tricky Project")
  })

  // it("Start button disappear after click", () => {
  //   const button = appWrapper.find("button")
  //   button.simulate("click")
  //   expect(appWrapper.find("button").exists()).toBeFalsy()
  // })
})