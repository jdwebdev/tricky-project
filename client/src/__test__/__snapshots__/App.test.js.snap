// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component renders correctly 1`] = `
<div
  className="App"
>
  <header>
    <h1>
      Tricky Project
    </h1>
  </header>
  <Game
    finishGame={[Function]}
  />
</div>
`;
